File: app/src/main/java/pl/srw/billcalculator/bill/calculation/TauronG11CalculatedBill.java
package pl.srw.billcalculator.bill.calculation;

import java.math.BigDecimal;

import pl.srw.billcalculator.pojo.ITauronPrices;

public class TauronG11CalculatedBill extends TauronCalculatedBill {

    private final int consumption;

    private final BigDecimal energiaElektrycznaNetCharge;
    private final BigDecimal oplataDystrybucyjnaZmiennaNetCharge;

    private final BigDecimal energiaElektrycznaVatCharge;
    private final BigDecimal oplataDystrybucyjnaZmiennaVatCharge;

    public TauronG11CalculatedBill(final int readingFrom, final int readingTo, final String dateFrom, final String dateTo, final ITauronPrices prices) {
        super(dateFrom, dateTo, prices.getOplataAbonamentowa(), prices.getOplataPrzejsciowa(), prices.getOplataDystrybucyjnaStala());
        consumption = readingTo - readingFrom;

        energiaElektrycznaNetCharge = countNetAndAddToSum(prices.getEnergiaElektrycznaCzynna(), consumption);
        oplataDystrybucyjnaZmiennaNetCharge = countNetAndAddToSum(prices.getOplataDystrybucyjnaZmienna(), consumption);

        energiaElektrycznaVatCharge = countVatAndAddToSum(energiaElektrycznaNetCharge);
        oplataDystrybucyjnaZmiennaVatCharge = countVatAndAddToSum(oplataDystrybucyjnaZmiennaNetCharge);
    }

    @Override
    public int getTotalConsumption() {
        return consumption;
    }

    @Override
    public BigDecimal getSellNetCharge() {
        return round(energiaElektrycznaNetCharge);
    }

    @Override
    public BigDecimal getSellVatCharge() {
        return round(energiaElektrycznaVatCharge);
    }
}

File: app/src/main/kotlin/pl/srw/billcalculator/bill/calculation/TauronG11CalculatedBill.kt
package pl.srw.billcalculator.bill.calculation

import org.threeten.bp.LocalDate
import pl.srw.billcalculator.pojo.ITauronPrices

class TauronG11CalculatedBill(readingFrom: Int, readingTo: Int, dateFrom: LocalDate, dateTo: LocalDate, prices: ITauronPrices)
    : TauronCalculatedBill(dateFrom, dateTo, prices.oplataAbonamentowa, prices.oplataPrzejsciowa, prices.oplataDystrybucyjnaStala) {

    override val totalConsumption = readingTo - readingFrom
    val consumptionFromJuly16 = countConsumptionPartFromJuly16(dateFrom, dateTo, totalConsumption)

    val energiaElektrycznaNetCharge = countNetAndAddToSum(prices.energiaElektrycznaCzynna, totalConsumption)
    val oplataDystrybucyjnaZmiennaNetCharge = countNetAndAddToSum(prices.oplataDystrybucyjnaZmienna, totalConsumption)
    val oplataOzeNetCharge = countNetAndAddToSum(prices.oplataOze, consumptionFromJuly16)

    val energiaElektrycznaVatCharge = countVatAndAddToSum(energiaElektrycznaNetCharge)
    val oplataDystrybucyjnaZmiennaVatCharge = countVatAndAddToSum(oplataDystrybucyjnaZmiennaNetCharge)
    val oplataOzeVatCharge = countVatAndAddToSum(oplataOzeNetCharge)

    override val sellNetCharge = energiaElektrycznaNetCharge.round()

    override val sellVatCharge = energiaElektrycznaVatCharge.round()
}